edgelet/support-bundle/src/runtime_util.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use std::io::Write;
+
+use failure::Fail;
+use futures::StreamExt;
+
+use edgelet_core::{LogOptions, Module, ModuleRuntime};
+
+use crate::error::{Error, ErrorKind};
+
+pub async fn get_modules(
+    runtime: &impl ModuleRuntime,
+    include_ms_only: bool,
+) -> Result<Vec<String>, Error> {
+    const MS_MODULES: &[&str] = &["edgeAgent", "edgeHub"];
+
+    let runtime_modules = runtime
+        .list_with_details()
+        .await
```

Comment:
`list_with_details()` returns a `Result`. Should this be `await?`? Otherwise, errors get silently ignored.

Same for line 41 below.

Reply:
I think this is a problem with the interface?

```
 async fn list(&self) -> Result<Vec<Self::Module>, Self::Error>;
 async fn list_with_details(&self) -> Result<(Self::Module, ModuleRuntimeState), Self::Error>;
```

Shouldn't `list_with_details` return `Vec<(Self::Module, ModuleRuntimeState)>`?

Reply:
I think it passes cargo check b/c the Result has .into_iter(), but it will give the wrong values

Reply:
Yeah, if you get an Err(_), that's silently ignored since into_iter() changes it to None.

Reply:
Yea, but I think the trait also needs to be changed to 
```
 async fn list_with_details(&self) -> Result<**Vec<**(Self::Module, ModuleRuntimeState)**>**, Self::Error>;
```

Reply:
The current master has it as:

```
Stream<Item = (Self::Module, ModuleRuntimeState), Error = Self::Error> + Send;
```

I must have accidentally changed it. Oops. Could you change it back to whatever makes sense? If you thing Vec<> works better than Stream<>, feel free to make it that.